components/chat/ToolCallApproval.tsx

```
@@ -0,0 +1,101 @@
+  DialogDescription,
+  DialogFooter,
+  DialogHeader,
+  DialogTitle,
+} from "@/components/ui/dialog";
+import { ScrollArea } from "@/components/ui/scroll-area";
+import { Check, X } from "lucide-react";
+
+interface ToolCallApprovalProps {
+  toolCall: {
+    name: string;
+    args: Record<string, any>;
```

ESLint

[error] Unexpected any. Specify a different type.

(@typescript-eslint/no-explicit-any)